File: programmers/level1/두 개 뽑아서 더하기/solution.kt
fun solution(numbers: IntArray): IntArray {
    val list = numbers.toList()
    return list.withIndex().flatMap { i -> list.withIndex().map { j -> i to j} }
        .filter { it.first.index != it.second.index }
        .map {it.first.value + it.second.value}
        .toSortedSet()
        .toIntArray()
}

File: programmers/level1/두 개 뽑아서 더하기/solution.js
/**
 *
 * @param {number[]} numbers
 * @returns
 */
function solution(numbers) {
  const result = [];
  return Array.from(
    new Set(
      numbers.flatMap((iv, i) =>
        numbers
          .map((jv, j) => i !== j && { iv, i, jv, j })
          .filter((item) => item)
          .map((item) => item.iv + item.jv)
      )
    )
  ).sort((a, b) => a - b);
}
